src/components/user-feed.jsx

```diff
@@ -3,21 +3,31 @@ import React from 'react';
 import PaginatedView from './paginated-view';
 import Feed from './feed';
 
-export default props => (
-  <div>
-    {props.viewUser.blocked ? (
-      <div className="box-body">
-        <p>You have blocked <b>{props.viewUser.screenName}</b>, so all of their posts and comments are invisible to you.</p>
-        <p><a onClick={()=>props.userActions.unban({username: props.viewUser.username, id: props.viewUser.id})}>Un-block</a></p>
-      </div>
-    ) : props.viewUser.isPrivate === '1' && !props.viewUser.subscribed && !props.viewUser.isItMe ? (
-      <div className="box-body">
-        <p><b>{props.viewUser.screenName}</b> has a private feed.</p>
-      </div>
-    ) : (
-      <PaginatedView {...props}>
-        <Feed {...props} isInUserFeed={true}/>
-      </PaginatedView>
-    )}
-  </div>
-);
+export default props => {
+  const {viewUser, location: {query}} = props;
+  const isBlocked = viewUser.blocked;
+  const isPrivate = viewUser.isPrivate === '1' && !viewUser.subscribed && !viewUser.isItMe;
+  const amIBlocked = viewUser.isPrivate === '0' && viewUser.statistics.posts !== '0' && props.visibleEntries.length === 0 && (!('offset' in query) || query.offset === '0');
```

Comment:
isPrivate is a string, '0' or '1'. So I think it is better to explicitly compare it with string.